src/utils/thread_helper.cpp

```diff
@@ -1016,19 +1025,23 @@ otError ThreadHelper::RetrieveTelemetryData(Mdns::Publisher *aPublisher, threadn
             uint8_t              len = sizeof(data);
             std::vector<uint8_t> networkData;
 
-            SuccessOrExit(error = otNetDataGet(mInstance, /*stable=*/false, data, &len));
-            networkData = std::vector<uint8_t>(&data[0], &data[len]);
-            wpanTopoFull->set_network_data(std::string(networkData.begin(), networkData.end()));
+            if (otNetDataGet(mInstance, /*stable=*/false, data, &len))
```

Comment:
Seems `otNetDataGet` returns an error instead of a boolean.

Reply:
Done.